src/com/goide/inspections/GoStructInitializationInspection.java

```diff
@@ -55,23 +56,42 @@ protected GoVisitor buildGoVisitor(@NotNull ProblemsHolder holder, @NotNull Loca
     return new GoVisitor() {
 
       @Override
-      public void visitCompositeLit(@NotNull GoCompositeLit compositeLit) {
-        GoLiteralValue literalValue = compositeLit.getLiteralValue();
+      public void visitLiteralValue(@NotNull GoLiteralValue literalValue) {
         GoStructType structType = getStructType(literalValue);
         if (structType == null || !isStructImportedOrLocalAllowed(structType, literalValue)) return;
 
-        List<String> elementsNames = getNames(literalValue.getElementList());
-        if (hasNull(elementsNames.toArray()) && areElementsNamesMatchesDefinitions(elementsNames, getFieldDefinitionsNames(structType))) {
-          holder.registerProblem(literalValue, "Unnamed field initializations", ProblemHighlightType.GENERIC_ERROR_OR_WARNING, QUICK_FIX);
-        }
+        List<GoElement> elements = literalValue.getElementList();
+        List<String> elementsNames = getNames(elements);
+        if (!areElementsNamesMatchesDefinitions(elementsNames, getFieldDefinitionsNames(structType))) return;
+
+        zip(elementsNames, elements).forEach(pair -> {
+          if (pair.first != null) return;
+          holder.registerProblem(pair.second, "Unnamed field initializations", ProblemHighlightType.WEAK_WARNING, QUICK_FIX);
+        });
       }
     };
   }
 
   @Nullable
   @Contract("null -> null")
-  private static GoStructType getStructType(@Nullable GoLiteralValue literal) {
-    return literal != null ? ObjectUtils.tryCast(GoPsiImplUtil.getLiteralType(literal.getParent(), false), GoStructType.class) : null;
+  private static GoStructType getStructType(@Nullable GoLiteralValue literalValue) {
+    GoCompositeLit lit = PsiTreeUtil.getParentOfType(literalValue, GoCompositeLit.class);
+    GoType litType = lit != null ? lit.getType() : null;
+    return isSliceElementValue(literalValue, litType)
```

Comment:
Example does not compile (should be `S: S{2, 3<caret>}` I guess), but fixed version works. Maybe we should not highlight such cases at all? (when there is no composite literal for literal value).